java-core/google-cloud-core/src/main/java/com/google/cloud/testing/StartupLogRecorder.java

```diff
@@ -21,18 +21,27 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-public class StartupLogAggregator {
+/**
+ * This class records the logs in memory and flush them onto the {@link java.util.logging.Logger}
```

Comment:
Can you add why/when this is helpful? (the reason why you had to write this.)

Reply:
This seems like a standalone component so i have just described 'record' and 'flush' part here. With regards to "why?" part, I've put reasoning in `BlockingProcessStreamReader.java` where this recorder is required. Let me know if that works.

Reply:
The current javadoc just explains implementation (what it does). It's failing to explain why it's helpful for "Startup" (as in the class name).

Reply:
Thanks for pointing this out @suztomo ,  i agree startup is not the right word for this independent component when it can be used in other non-startup use cases. I've renamed it, and also put that as an example in javadoc.

Reply:
fixed and hence resolving this.